### PR TITLE
Determine check 2

### PR DESCRIPTION
**Pull request by ezschemi**
Implemented check-checking.

### CODE REVIEW

**Review by rongg — APPROVED** [review]

**Comment by fleawig** [comment]
Great! This adds the before-move prohibition on moving self into check, and adds the "You are in check" to the board afterwards (which we can then change to integrate better into UI.)

**Comment by fleawig** [comment]
I believe you need to resolve the conflicts before approval will be enabled (it's grayed out as of now.)

**Review by fleawig — APPROVED** [review]

**Pull request merged**
